bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeHandler.java

```diff
@@ -0,0 +1,439 @@
+                case MODE_COOL:
+                    mode = GREE_MODE_COOL;
+                    break;
+                case MODE_HEAT:
+                    mode = GREE_MODE_HEAT;
+                    break;
+                case MODE_DRY:
+                    mode = GREE_MODE_DRY;
+                    break;
+                case MODE_FAN:
+                case MODE_FAN2:
+                    mode = GREE_MODE_FAN;
+                    break;
+                case MODE_ECO:
+                    // power saving will be set after the uinit was turned on
+                    mode = GREE_MODE_COOL;
+                    break;
+                case MODE_ON:
+                case MODE_OFF:
+                    device.setDevicePower(socket, modeStr.equals(MODE_ON) ? 1 : 0);
+                    break;
+                default:
+                    if (isNumber) {
+                        // Support selecting the mode by number, maybe specific models support
+                        // additional modes
+                        mode = Integer.parseInt(modeStr);
+                    } else {
+                        logger.debug("Invalid mode requested: {}", command);
+                    }
+                    break;
+            }
+            logger.debug("Mode {} mapped to {}", modeStr, mode);
+        }
+
+        if (mode != -1) {
+            // Turn on the unit if currently off
+            if (!isNumber && (device.getIntStatusVal("Pow") == 0)) {
+                logger.debug("Send Auto-ON for mode {}", mode);
+                device.setDevicePower(socket, 1);
+            }
+
+            // Select mode
+            logger.debug("Select mode {}", mode);
+            device.SetDeviceMode(socket, mode);
+
+            // Check for secondary action
+            switch (modeStr) {
+                case MODE_ECO:
+                    // Turn on power saving for eco mode
+                    logger.debug("Turn on Power-Saving");
+                    device.setDevicePwrSaving(socket, 1);
+                    break;
+                case MODE_TURBO:
+                    device.setDeviceTurbo(socket, 1);
+                    break;
+            }
+        }
+    }
+
+    private boolean isMinimumRefreshTimeExceeded() {
+        long currentTime = System.currentTimeMillis();
+        long timeSinceLastRefresh = currentTime - lastRefreshTime;
+        if (timeSinceLastRefresh < MINIMUM_REFRESH_TIME_MS) {
+            return false;
+        }
+        lastRefreshTime = currentTime;
+        return true;
+    }
+
+    private void startAutomaticRefresh() {
+
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+
+                try {
+                    logger.debug("Executing automatic update of values");
+                    // safeguard for multiple REFRESH commands
+                    if (isMinimumRefreshTimeExceeded()) {
+                        logger.debug("Fetching status values from device.");
```

Comment:
Can this be replaced by using the debugger?

Reply:
what do you mean?

Reply:
I want to see it when users are providing logs

Reply:
removed